src/rainfly/src/lib/utils/file-utils.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @fileoverview Common utility function for working with files
+ */
+
+import JSZip from 'jszip';
+
+/**
+ * Fetch a file from the given URL and return name and text data
+ * @param {string} url - URL of the file to fetch
+ * @return {Promise<{name: string, data: string}>} - Promise that resolves to
+ * an object with the name and data of the file
+ */
+export async function fetchTextFile(url) {
+  let filename = url.split('/').pop();
```

Comment:
I don't see any error handling here. Is `jszip` handling errors somehow?

Reply:
@hoch Do you mean error handling for the filename handling or the fetch requests (next few lines)? `jszip` isn't involved in this part of the code. I've gone ahead and wrapped the fetch requests in a try catch.

Reply:
Sorry for late response. Looks good!